### PR TITLE
Log Prob Visualiser

### PR DESCRIPTION
**Pull request by neelnanda-io**
Added a log prob visualizer! This shows you some text, with tokens coloured by the log prob of the model. If you hover over a token, it shows you the top 10 tokens predicted, along with logits and log probs, and the values + rank of the correct token. The correct token is highlighted in the list of top 10 if it occurs in there (if so, it appears twice).

image

image

### CODE REVIEW

**Comment by neelnanda-io** [comment]
Did some internal refactoring of the Token class - pulled out the formatting text part into a separate function, and added a `TokenCustomTooltip` class that can display an arbitrary React Component as its tooltip.

I also have some local code that automates the creation of BS boilerplate (for writing the visualization and for mocking)

**Review by alan-cooney — CHANGES_REQUESTED** [review]
A few small changes in the comments - note the checks are also failing so there may be an issue still after these are made

File: react/src/repl_tf.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+const rl = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout
+});
+
+console.log('Welcome to the TensorFlow.js REPL!');
+console.log('Enter a TensorFlow.js expression or statement and press enter to evaluate it.');
+
+rl.on('line', (input) => {
+    try {
```
[the remark this answers is not in the document]
Comment (neelnanda-io) [reply react/src/repl_tf.js:29]: Oh lol, nope.

**Comment by neelnanda-io** [comment]
Not sure how the dependency changes happened!

**Comment by neelnanda-io** [comment]
All changes made

**Comment by neelnanda-io** [comment]
I did not notice that there were tests! Turns out I had an actual bug, so good shout (imported some things twice in index.ts), fixed.
Also funky, my local code thought the library was v 1.6.0 not 1.34.0, which broke the topk test. I added a thing which monkey patches the version to avoid each update breaking this, but happy to revert if you want. It _did_ catch a real bug in my case, but will also break for each new version.
